src/core_modules/capture-core/hooks/useTrackedEntityTypesWithCorrelatedPrograms.js

```diff
@@ -32,9 +33,28 @@ export const useTrackedEntityTypesWithCorrelatedPrograms = (): TrackedEntityType
                 },
                 searchGroups,
                 enrollment,
+                stages,
+                useFirstStageDuringRegistration,
             }: TrackerProgram) => {
                 const accumulatedProgramsOfTrackedEntityType =
                   acc[trackedEntityTypeId] ? acc[trackedEntityTypeId].programs : [];
+                const programMetadata = { programId, programName, searchGroups, enrollment };
+                if (useFirstStageDuringRegistration) {
+                    const accessibleStages = [...stages.values()].filter(({ access }) => access.write);
+                    if (accessibleStages.length) {
+                        const firstAccessibleStage = accessibleStages[0];
```

Comment:
Even if it seems that the order of our stages is correct, this order is really important to show the correct stage on the registration page. Perhaps we should rewrite this to use the `sortOrder`-flag on the stage itself?